### ISSUE TITLE
Can't get useContext example to work in react

### ISSUE THREAD

**Issue by PerfectionCSGO** [mobxjs/mobx-react-lite#118]
In mobx-react with with class components I was using my case like so:
```
@connect('someStore')
@obsevable
class AppTile extends ConnectedComponent<IProps,IRootStore,IState> {} //example
```
But now I'm trying to get mobx-react-lite to work in my test.
I'm using the useContext example provided like so:

```
const AppTile: React.FC<AppTileProps> = ({ applicationId }: AppTileProps) => {
    const applicationStore = React.useContext(ApplicationStoreContext);
    const application = applicationStore.getApplication(applicationId);
    application.fetch();

    return (<AppTileRenderer state={application.state} />);
  };
```

I'm using MobX dev tools and I can indeed observe that the state does change and that mobx does things. However, this is not reflected in my component tree. The value never changes.

I added the usage of observer() around AppTile trying to understand what I'm doing wrong but then react will complain like so:

```
Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.

    at invariant (react\cjs\react.development.js:88:15)
    at resolveDispatcher (react\cjs\react.development.js:1436:28)
    at Object.useState (react\cjs\react.development.js:1461:20)
    at useForceUpdate (mobx-react-lite\dist\index.js:141:31)
    at useObserver (mobx-react-lite\dist\index.js:156:27)
    at wrappedComponent (mobx-react-lite\dist\index.js:196:20)
    at renderWithHooks (webpack-internal:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js:12939:18)
    at updateFunctionComponent (webpack-internal:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js:14628:20)
    at updateSimpleMemoComponent (webpack-internal:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js:14574:10)
    at updateMemoComponent (webpack-internal:///./node_modules/@hot-loader/react-dom/cjs/react-dom.development.js:14493:14)
```

Could I possibly get an example of how to use mobx-react-lite?

I'm using teact 16.8.6,  mobx 5.9.4 and mobx-react-lite 1.2.0 with typescript 3.4.1. So if this is a bleeding edge case then I understand.

**Comment by FredyC** [comment]
Please provide a working (or broken) example ideally in CodeSandbox. It's hard to follow your train of thoughts here.

**Comment by PerfectionCSGO** [comment]
> Please provide a working (or broken) example ideally in CodeSandbox. It's hard to follow your train of thoughts here.

I apologize I tried to do it in codesandbox but I'm not used to this kind of boilerplating and couldn't get it to work.

I build my own boilerplate some time ago for personal use and build my situation in there.  However, it works.

https://github.com/perfectioncsgo/perfection-tv

What I have learned that the fetch function was causing an infinite loop. For some reason in my test project it's causing react hook errors. While in my boilerplate it just freezes in the browser.

```
const AppTile: React.FC<AppTileProps> = ({ applicationId }: AppTileProps) => {
    const applicationStore = React.useContext(ApplicationStoreContext);
    const application = applicationStore.getApplication(applicationId);
    application.fetch(); // is an action and will cause infinite loops

    return (<AppTileRenderer state={application.state} />);
  };
  export default observer(AppTile);
```

tl;dr I think you can close this question as it's a localized problem. But an example of how to properly use mobx-react-lite would be nice. I'm showing an example in my boilerplate with 5 items. But I have a list of 100.000 async loaded items, but ofcourse I'm using a hoc to load/unload appropriately in a real world scenario.

**Comment by FredyC** [comment]
> But an example of how to properly use mobx-react-lite would be nice.

I would say there is a fair amount of examples in README :) Either way, please follow #115 for more examples coming.
